### ISSUE TITLE
compile wapi

### ISSUE THREAD

**Issue by vsivanagulu** [vy/wapi#1]
i dont know how to run this can u please tell me how to compile this wapi for arm

**Comment by vy** [comment]
I have not been using (and hence, maintaining) WAPI for ~6 years. It needs some *love* to make it work with the most recent libnl API:
```
$ sudo aptitude install -R libnl-3-dev libiw-dev
$ gcc -shared -fno-strict-aliasing -DLIBNL1 -lnl -lm -liw -Iinclude -Isrc -I/usr/include/libnl3 src/util.c src/network.c src/wireless.c -o /tmp/libwapi.so
src/wireless.c:854:26: error: conflicting types for ‘nl_socket_alloc’
 static struct nl_handle *nl_socket_alloc(void)
                          ^
In file included from /usr/include/libnl3/netlink/netlink.h:34:0,
                 from /usr/include/libnl3/netlink/genl/genl.h:15,
                 from src/wireless.c:12:
/usr/include/libnl3/netlink/socket.h:22:25: note: previous declaration of ‘nl_socket_alloc’ was here
 extern struct nl_sock * nl_socket_alloc(void);
                         ^
src/wireless.c: In function ‘nl_socket_alloc’:
src/wireless.c:856:9: warning: implicit declaration of function ‘nl_handle_alloc’ [-Wimplicit-function-declaration]
  return nl_handle_alloc();
         ^
src/wireless.c:856:9: warning: return makes pointer from integer without a cast [-Wint-conversion]
src/wireless.c: At top level:
src/wireless.c:860:13: error: conflicting types for ‘nl_socket_free’
 static void nl_socket_free(struct nl_sock *h)
             ^
In file included from /usr/include/libnl3/netlink/netlink.h:34:0,
                 from /usr/include/libnl3/netlink/genl/genl.h:15,
                 from src/wireless.c:12:
/usr/include/libnl3/netlink/socket.h:24:14: note: previous declaration of ‘nl_socket_free’ was here
 extern void  nl_socket_free(struct nl_sock *);
              ^
src/wireless.c: In function ‘nl_socket_free’:
src/wireless.c:862:2: warning: implicit declaration of function ‘nl_handle_destroy’ [-Wimplicit-function-declaration]
  nl_handle_destroy(h);
  ^
src/wireless.c: In function ‘nl80211_cmd_handler’:
src/wireless.c:940:19: warning: passing argument 1 of ‘genl_connect’ from incompatible pointer type [-Wincompatible-pointer-types]
  if (genl_connect(sock))
                   ^
In file included from src/wireless.c:12:0:
/usr/include/libnl3/netlink/genl/genl.h:23:13: note: expected ‘struct nl_sock *’ but argument is of type ‘struct nl_handle *’
 extern int  genl_connect(struct nl_sock *);
             ^
src/wireless.c:948:35: warning: passing argument 1 of ‘genl_ctrl_resolve’ from incompatible pointer type [-Wincompatible-pointer-types]
  ret = family = genl_ctrl_resolve(sock, "nl80211");
                                   ^
In file included from src/wireless.c:14:0:
/usr/include/libnl3/netlink/genl/ctrl.h:30:14: note: expected ‘struct nl_sock *’ but argument is of type ‘struct nl_handle *’
 extern int   genl_ctrl_resolve(struct nl_sock *,
              ^
src/wireless.c:1004:30: warning: passing argument 1 of ‘nl_send_auto_complete’ from incompatible pointer type [-Wincompatible-pointer-types]
  ret = nl_send_auto_complete(sock, msg);
                              ^
In file included from /usr/include/libnl3/netlink/genl/genl.h:15:0,
                 from src/wireless.c:12:
/usr/include/libnl3/netlink/netlink.h:66:14: note: expected ‘struct nl_sock *’ but argument is of type ‘struct nl_handle *’
 extern int   nl_send_auto_complete(struct nl_sock *,
              ^
src/wireless.c:1026:39: warning: passing argument 1 of ‘nl_recvmsgs’ from incompatible pointer type [-Wincompatible-pointer-types]
  for (ret = 1; ret > 0; ) nl_recvmsgs(sock, cb);
                                       ^
In file included from /usr/include/libnl3/netlink/genl/genl.h:15:0,
                 from src/wireless.c:12:
/usr/include/libnl3/netlink/netlink.h:77:14: note: expected ‘struct nl_sock *’ but argument is of type ‘struct nl_handle *’
 extern int   nl_recvmsgs(struct nl_sock *, struct nl_cb *);
              ^
```
As always, patches/contributions are welcome.

**Comment by vy** [comment]
Hopefully fixed in #2.
